### PR TITLE
libtar: drop Windows (not only msvc) + fix test package + modernize

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by madebr — APPROVED** [review]

**Comment by SpaceIm** [comment]
Funny, test package with apple-clang 12 shared only. I guess that it's the first time that libar is compiled with apple-clang 12 in CCI.

**Review by madebr — DISMISSED** [review]

**Comment by SpaceIm** [comment]
Now the same configuration succeeds, but another one fails. Something is not reproducible in test package.

**Comment by madebr** [comment]
Weird how `88888c4c42b4cc249e5034e057b08d49798b4b27` is shown once in
https://c3i.jfrog.io/c3i/misc/summary.html?json=https://c3i.jfrog.io/c3i/misc/logs/pr/6395/3/libtar/1.2.20//summary.json
and twice in 
https://c3i.jfrog.io/c3i/misc/summary.html?json=https://c3i.jfrog.io/c3i/misc/logs/pr/6395/4/libtar/1.2.20//summary.json

**Comment by SpaceIm** [comment]
Oh, maybe that's the issue: libtar can't handle PAX format, and PAX format is default format of tarfile since python 3.8

**Review by madebr — APPROVED** [review]

File: recipes/libtar/all/test_package/conanfile.py [hunk 1]
```diff
@@ -14,7 +14,7 @@ def build(self):
         cmake.build()
 
     def test(self):
-        with tarfile.open("test.tar", "w") as f:
+        with tarfile.open("test.tar", "w", format=tarfile.GNU_FORMAT) as f:
```
Comment (SSE4) [inline recipes/libtar/all/test_package/conanfile.py:17]: sounds reasonable

**Review by SSE4 — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**